src/vnet/interface.h

```diff
@@ -788,8 +793,13 @@ typedef enum
   _(RX_NO_BUF, rx-no-buf, if)			\
   _(RX_MISS, rx-miss, if)			\
   _(RX_ERROR, rx-error, if)			\
-  _(TX_ERROR, tx-error, if)         \
-  _(MPLS, mpls, if)
+  _(TX_ERROR, tx-error, if)                     \
+  _(MPLS, mpls, if)                             \
+  _(IS_UP, up, if)                              \
+  _(IS_IP4, is-ip4, if)                         \
+  _(IS_IP6, is-ip6, if)                         \
+  _(L3MTU, l3mtu, if)                           \
```

Comment:
those values do not belong into the interface counters. They are not statistics values, but interface state values. The binary API should be used to read them, not the stats segment.

Reply:
VPP not only exporting statistics(like counters) but also name vectors or buffer sizes. Those are also state values.
Another thing why it's here that I've tried to avoid usage of binapi for monitoring as much as it's possible. Thing is, that monitoring will constantly fetch metrics from vpp. This will lead to repetitive binapi calls. Stats segment in this case is more lightweight and stand-alone. This stats will be written to corresponding counters so collector will just fetch via mapped memory segment.